server/komodo-server/src/main/java/org/komodo/rest/connections/SyndesisConnectionSynchronizer.java

```diff
@@ -94,14 +94,23 @@ private void handleAddConnection(EventMsg event, boolean update) throws KExcepti
     private void handleDeleteConnection(EventMsg event) throws KException {
         // note here that the datasource is already deleted from the syndesis
         // so we would need to search by local cached event id
-        DefaultSyndesisDataSource sds = this.openshiftClient.getSyndesisDataSourceById(bogusCredentials,
-                event.getId());
+        DefaultSyndesisDataSource sds = this.openshiftClient.getSyndesisDataSourceById(event.getId());
         if (sds != null) {
             deleteConnection(sds);
         }
     }
 
     public void addConnection(DefaultSyndesisDataSource sds, boolean update) {
+        try {
+            // this is avoid circular creation of the virtualization connection that is published through syndesis
+            DataVirtualization dv = kengine.getWorkspaceManager().findDataVirtualization(sds.getSyndesisName());
+            if (dv != null && dv.getId().equals(sds.getId())) {
```

Comment:
These are not the right checks.  You should not lookup the datavirtualization by the syndesisname because that may not be stable - it should be found by the source id matching the DefaultSyndesisDataSource id.

We should probably rename DefaultSyndesisDataSource.getId as getSyndesisId - as it is the syndesis assigned id.  

dv.getId on the other hand returns the teiid-komodo generated id, so an equality check won't work.

Reply:
yes, that was mistake, I started with corrected in other places, looks like I missed here.